app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.kt

```diff
@@ -0,0 +1,163 @@
+package com.quran.labs.androidquran.database
+
+import android.content.ContentValues
+import android.content.Context
+import android.database.Cursor
+import android.database.sqlite.SQLiteDatabase
+import android.util.SparseArray
+import androidx.annotation.WorkerThread
+import com.quran.labs.androidquran.common.LocalTranslation
+import com.quran.labs.androidquran.dao.translation.TranslationItem
+import com.quran.labs.androidquran.database.TranslationsDBHelper.TranslationsTable
+import com.quran.labs.androidquran.util.QuranFileUtils
+import timber.log.Timber
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class TranslationsDBAdapter @Inject constructor(
+  private val context: Context,
+  adapter: TranslationsDBHelper,
+  private val quranFileUtils: QuranFileUtils
+) {
+  private val db: SQLiteDatabase = adapter.writableDatabase
+
+  @Volatile
+  private var cachedTranslations: List<LocalTranslation>? = null
+
+  var lastWriteTime: Long = 0
+    private set
+
+  fun getTranslationsHash(): SparseArray<LocalTranslation> {
+    val result = SparseArray<LocalTranslation>()
+    for (item in translations) {
+      result.put(item.id, item)
+    }
+    return result
+  }
+
+  // intentional, since cachedTranslations can be replaced by another thread, causing the check
+  // to be true, but the cached object returned to be null (or to change).
+  @get:WorkerThread
+  val translations: List<LocalTranslation>
```

Comment:
let's make it a function instead

Reply:
> minor

yes brother, will make sure of that in the future

Reply:
i have push this change, and it needs changes of variable in this l[ine](https://github.com/wiryadev/quran_android/blob/bb9cb39c07d4ff0526a598c6051eb2b58236cc1e/app/src/main/java/com/quran/labs/androidquran/presenter/translation/BaseTranslationPresenter.kt#L180)

from
`Single.fromCallable<List<LocalTranslation>> { translationsAdapter.translations }`

to
`Single.fromCallable<List<LocalTranslation>> { translationsAdapter.getTranslations()}`